independent-projects/vertx-java-io/pom.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../parent/pom.xml</relativePath>
+    </parent>
+
+    <groupId>io.quarkus.vertx-java-io</groupId>
+    <artifactId>quarkus-vertx-java-io</artifactId>
+    <name>Ancillary classes for using Vert.x with frameworks designed to read/write from/to java.io streams</name>
```

Comment:
I'm not so sure we should create an independent project for this kind of stuff. AFAIK there are some drawbacks related to version management, release process, etc.

@gsmet may know more.

Reply:
Also `quarkus-vertx-java-io` feels really weird. I'd call it `quarkus-vertx-buffer`.

Reply:
> I'm not so sure we should create an independent project for this kind of stuff. AFAIK there are some drawbacks related to version management, release process, etc. @gsmet may know more.

+1 about the concern about the version management. I was hoping all was automated given that there are several independent projects already. Anyway, I am open to discuss where the module should be hosted. WDYT @gsmet?

> Also `quarkus-vertx-java-io` feels really weird. I'd call it `quarkus-vertx-buffer`.

Re-using the `VertxBufferImpl` was rather a secondary idea. The module was primarily intended to host the `VertxOutputStream` and perhaps also other java.io stream implementations usable for old frameworks wanting to run on top of Vert.x. I do not care much about the module name. `quarkus-vertx-java-io`, `quarkus-vertx-buffer` or even `quarkus-vertx-web-utils` would work for me.

Reply:
> Re-using the VertxBufferImpl was rather a secondary idea. 

In that case something like `quarkus-vertx-utils` might make more sense...

BTW I also found `io.quarkus.resteasy.runtime.standalone.VertxOutputStream`. Isn't it a copy of the original `VertxOutputStream`?

Reply:
> > Re-using the VertxBufferImpl was rather a secondary idea.
> 
> In that case something like `quarkus-vertx-utils` might make more sense...

OK, let me rename it to `quarkus-vertx-utils` unless nobody has a better idea.

> BTW I also found `io.quarkus.resteasy.runtime.standalone.VertxOutputStream`. Isn't it a copy of the original `VertxOutputStream`?

They are different. The VertxOutputStream from resteasy-reactive-vertx extends java.io.OutputStream rather than org.jboss.resteasy.spi.AsyncOutputStream

Reply:
> I'm not so sure we should create an independent project for this kind of stuff. AFAIK there are some drawbacks related to version management, release process, etc.

I can't remember there being any such drawbacks

Reply:
> > I'm not so sure we should create an independent project for this kind of stuff. AFAIK there are some drawbacks related to version management, release process, etc.
> 
> I can't remember there being any such drawbacks

There are some version properties in various independent projects, which have to be kept in sync on various places. This PR makes the mess a bit worse by adding two new properties in the new module.
It would be nice to hear what @gsmet thinks.